internal/dag/gatewayapi_processor.go

```diff
@@ -61,6 +63,34 @@ func (p *GatewayAPIProcessor) Run(dag *DAG, source *KubernetesCache) {
 
 	for _, listener := range p.source.gateway.Spec.Listeners {
 
+		var listenerSecret *Secret
+		var err error
+
+		// Check for TLS on the Gateway.
+		if tls := listener.TLS; tls != nil {
+			// Validate the CertificateRef is configured.
+			if tls.CertificateRef == nil {
+				p.Error(contour_api_v1.ConditionTypeTLSError, "SecretNotConfigured",
+					"Spec.VirtualHost.TLS Secret is not configured")
+				continue
+			}
+
+			// Validate a v1.Secret is referenced which can be kind: secret & group: core.
+			// ref: https://github.com/kubernetes-sigs/gateway-api/pull/562
+			if strings.ToLower(tls.CertificateRef.Kind) != "secret" && strings.ToLower(tls.CertificateRef.Group) != "v1" {
+				p.Error(contour_api_v1.ConditionTypeTLSError, "InvalidReference",
+					"Spec.VirtualHost.TLS Secret must be type v1.Secret")
+				continue
+			}
+
+			listenerSecret, err = p.source.LookupSecret(types.NamespacedName{Name: tls.CertificateRef.Name, Namespace: p.source.gateway.Namespace}, validSecret)
```

Comment:
Might not be a bad idea to expand the cert validation. You should be able to copy/paste [this code](https://github.com/projectcontour/contour-operator/pull/237/files#diff-feefa601b9dc749f80c19b5769691942730947a8a8226c6ff15767916e58980cR242-R271) I planned for the operator.

Reply:
Right now in Contour, certs are validated before they are put into the cache (https://github.com/projectcontour/contour/blob/main/internal/dag/secret.go#L34:6). This logic here in the processor is just checking to see if we've cached a matching secret to use or not. 

If we want we could expand the validation logic in a separate PR in the cache bits.